### DISCUSSION TITLE
Ubuntu 24.04 LTS Updates (mods necessary to run scripts)

### DISCUSSION THREAD

**Discussion by ptapper** [neutrinolabs/pulseaudio-module-xrdp#117] [General]
When running the install_pulseaudio_sources_apt_wrapper.sh in Ubuntu 24.04 LTS if you run into an error 
"** schroot failed. ..."
The log will tell you:
"E: You must put some 'deb-src' URIs in your sources.list" 
This is because Ubuntu source have moved to /etc/apt/source.list.d/ubuntu.sources
In order to run install_pulseaudio_sources_apt_wrapper.sh successfully you have to modify the script around line 280 (in commit d0a4c12221592995282d9b6f2a2fe226767839ad) as follow:
#Copy some files to the build root
 `for file in $(find /etc/apt/ /etc/apt/sources.list.d -maxdepth 1 -type f -name '*.sources'); do`

Before running the script again, you'll have to remove the old built root (if you tried to run the script already).
`sudo rm -r /var/lib/pa-build/root`

I hope this helps.

**Comment by matt335672** [comment]
Thanks @ptapper 

One question - are you using the latest version of the script from Git, or the release version? The Git version has a similar fix in it (see #103)

**Reply by ptapper** [reply]
@matt335672 
Yes, I used the latest version of the scripts from Git, using:
git clone https://github.com/neutrinolabs/pulseaudio-module-xrdp.git
The change I suggested is only for Ubuntu 24.04 LTS, I did not test other versions of Ubuntu or any other distros. 
In 24.04, .list extension got changed to .sources, scanning functionality - location, as mentioned in [https://github.com/neutrinolabs/pulseaudio-module-xrdp/pull/103](url) still works. 

Other information that may be relevant to your question:
xrdp is the release version downloaded using sudo apt-get install xrdp. xrdp version 0.9.24-4
xorgxrdp is the release version downloaded using sudo apt-get install xorgxrdp. xorgxrdp version 1:0.9.19-1
